pkgs/by-name/op/open-adventure/package.nix

```
@@ -0,0 +1,81 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitLab,
+  python3Packages,
+  pkg-config,
+  libedit,
+  cppcheck,
+  coreutils,
+  asciidoctor,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "open-adventure";
+  version = "1.20";
+  src = fetchFromGitLab {
+    owner = "esr";
+    repo = "open-adventure";
+    tag = finalAttrs.version;
+    hash = "sha256-xbsMz99CNLhpM6BSJVcRzxPB6tUYfPy/3Z+8BKt8b1E=";
+  };
+
+  nativeBuildInputs = [
+    python3Packages.python
+    pkg-config
+    asciidoctor
+  ];
+
+  buildInputs = [
+    python3Packages.pyyaml
+    libedit
+  ];
+
+  doCheck = true;
+
+  nativeCheckInputs = [
+    python3Packages.pylint
+    cppcheck
+  ];
+
+  postPatch = ''
+    patchShebangs --build make_dungeon.py
+
+    # https://gitlab.com/esr/open-adventure/-/issues/70
+    substituteInPlace Makefile --replace-fail "--template " "--template="
+
+    substituteInPlace tests/tapview --replace-fail "/bin/echo" ${lib.getExe' coreutils "echo"}
+  '';
+
+  buildFlags = ''
+    advent.6
+  '';
```

```suggestion
  buildFlags = [
    "advent.6"
  ];
```
`buildFlags` is a list.

---

Ahh ok ok